railties/lib/rails/test_unit/reporter.rb

```diff
@@ -15,8 +15,12 @@ def aggregated_results # :nodoc:
       filtered_results.reject!(&:skipped?) unless options[:verbose]
       filtered_results.map do |result|
         location, line = result.method(result.name).source_location
-        "bin/rails test #{location}:#{line}"
+        "bin/rails test #{relative_path_for(location)}:#{line}"
       end.join "\n"
     end
+
+    def relative_path_for(file)
+      file.sub(/^#{Dir.pwd}\/?/, '')
```

Comment:
@kaspth Have you tried [`Pathname#relative_path_from`](http://ruby-doc.org/stdlib-2.1.0/libdoc/pathname/rdoc/Pathname.html#method-i-relative_path_from)? I think it can do it for you.